### PR TITLE
#24: implement DatabaseManager and  connect() method with tests coverage

### PR DESCRIPTION
**Pull request by Shtramak**
connect to #24

### CODE REVIEW

File: src/main/java/juja/sqlcmd/DatabaseManager.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package juja.sqlcmd;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class DatabaseManager {
+    private Connection connection;
```
Comment (ivshapovalov) [inline src/main/java/juja/sqlcmd/DatabaseManager.java]: Class and Interface Declarations
http://www.oracle.com/technetwork/java/codeconventions-150003.pdf

File: src/test/java/juja/sqlcmd/DatabaseManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    @Test
+    public void testSuccessfulConnection() {
+        boolean expected = databaseManager.connect("sqlcmd", "sqlcmd", "sqlcmd");
+        assertEquals(expected, true);
+    }
+
+    @Test
+    public void testUnSuccessfulConnection() {
```
Comment (ivshapovalov) [inline src/test/java/juja/sqlcmd/DatabaseManagerTest.java]: I think 'Unsuccessful' is particular word. Decouple on two words is unnecessary maybe.

**Review by ivshapovalov — CHANGES_REQUESTED** [review]

File: src/main/java/juja/sqlcmd/DatabaseManager.java (hunk 1 above)
Comment (Shtramak) [reply src/main/java/juja/sqlcmd/DatabaseManager.java]: With this you mean to change the positions of "Class (static) variables" and "Instance variables"?

File: src/test/java/juja/sqlcmd/DatabaseManagerTest.java (hunk 2 above)
Comment (Shtramak) [reply src/test/java/juja/sqlcmd/DatabaseManagerTest.java]: Sure, thanks... And testFailedConnection() even beter name, I think. Will rename this test

File: src/main/java/juja/sqlcmd/DatabaseManager.java (hunk 1 above)
Comment (ivshapovalov) [reply src/main/java/juja/sqlcmd/DatabaseManager.java]: Exactly

File: src/test/java/juja/sqlcmd/DatabaseManagerTest.java [hunk 3]
```diff
@@ -3,25 +3,26 @@
...
-        assertEquals(expected, true);
+    public void testCorrectConnection() {
+        assertTrue(databaseManager.connect("sqlcmd", "sqlcmd", "sqlcmd"));
     }
 
     @Test
-    public void testUnSuccessfulConnection() {
-        boolean expected = databaseManager.connect("NotSqlcmd", "sqlcmd", "sqlcmd");
-        assertEquals(expected, false);
+    public void testConnectionWithWrongData() {
```
Comment (Shtramak) [inline src/test/java/juja/sqlcmd/DatabaseManagerTest.java]: Is it ok to write few asserts in one method if it tests exactly one functionality? I was thinking to separate these asserts, but finlly decide to leave them in one test method

Comment (ivshapovalov) [reply src/test/java/juja/sqlcmd/DatabaseManagerTest.java]: Try to name methods with original method name. For example:
testConnectWhen...Returns...()
or without 'test' word
connectWhen...ExecutedCorrectly()

About your question:
I think on these step it is needlessly.
You can create methods such as: 
connectWhenDatabaseNameIsIncorrectReturnsFalse()
connectWhenUserNameIsIncorrectReturnsFalse()
connectWhenPasswordIsIncorrectReturnsFalse()

**Review by AVM82 — APPROVED** [review]
Looks good at me

**Review by ivshapovalov — APPROVED** [review]

File: src/test/java/juja/sqlcmd/DatabaseManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Before
+    public void init() {
+        databaseManager = new DatabaseManager();
+    }
+
+    @Test
+    public void connectWhenValidDataReturnsTrue() {
+        assertTrue(databaseManager.connect("sqlcmd", "sqlcmd", "sqlcmd"));
+    }
```
Comment (kortov) [inline src/test/java/juja/sqlcmd/DatabaseManagerTest.java]: You could name it "connectWhenValidAllParametersReturnsTrue"

**Review by kortov — APPROVED** [review]

File: src/test/java/juja/sqlcmd/DatabaseManagerTest.java (hunk 4 above)
Comment (Shtramak) [reply src/test/java/juja/sqlcmd/DatabaseManagerTest.java]: It sounds realy better... Thanks!

**Review by adaleorion — APPROVED** [review]
looks good, thanks

**Pull request merged**